### ISSUE TITLE
Clean up classes and style tags in template files

### ISSUE THREAD

**Issue by liambaloh** [OneHourGameJam/OneHourGameJam#135]
The template files use style tags very poorly. A lot of that should be moved into site.css and nightmode.css

Additionally many of the pages were just copy-pasted - for example the jams, entries and authors pages are all copy-pastes of each other, editcontent is a copy-paste of main page, etc. this means there are many illogical and non-descriptive classes used in templates.

**Comment by mkalam-alami** [comment]
There is potential for cleaning up the HTML by using the [advanced Mustache features](https://github.com/bobthecow/mustache.php/wiki/Mustache-Tags):

* Lambdas (aka. filters), eg. for panels
* Partials (aka. includes), eg. for games

**Comment by liambaloh** [comment]
Correct me if I'm wrong, but wouldn't lambdas involve putting quite a bit of template-specific code into php scripts? Don't get me wrong, we do this in a few places as it was a quick way to implement or fix things, but I'd prefer to avoid any html or css being generated by php code, at least unless we add a template-specific php file into the templates directory, which could work.

I didn't really understand partials and how they could be useful to us though.

**Comment by mkalam-alami** [comment]
Yeah lambdas imply code, so ideally they shouldn't have too much HTML in them. I don't mind your suggestions to put them near the templates.

Partials are a bit like functions (except you can't explicitly pass params, just the full context), they could be useful mostly for displaying game cards or details consistently between pages, without having to copy-paste the code. They may also help with the page you've mentioned as duplicated.

This stuff shouldn't change in depth how the templates work, but it can help fix code duplication.

**Comment by liambaloh** [comment]
So would a partial allow us to, for example, have a card.html and then indicate that a bit of main.html and a bit of jam.html should instead be rendered with the card.html template instead? Can this be done in a way that only involves templates or do you need php code?

**Comment by mkalam-alami** [comment]
Basically when initializing Mustache we'd declare the folder of all partials (eg. `templates/partials`), then assuming we have a `templates/partials/card.html` file, we could simply use it with:

```html
{{>card}}
```

The only trick is that all the variables used within the `card.html` file (typically here, a game) must already be set in the context of the template.

**Comment by mkalam-alami** [comment]
If no one beats me to it I'll give it a try at some point :)

**Comment by liambaloh** [comment]
Soooo, have you been able to get it to work? I've tried the following

index.php:
![image](https://user-images.githubusercontent.com/1364475/52907573-2d4e5880-325c-11e9-9f6b-9f05a589dafd.png)

and header.html
![image](https://user-images.githubusercontent.com/1364475/52907581-3dfece80-325c-11e9-8182-488212167640.png)

and the result was:
![image](https://user-images.githubusercontent.com/1364475/52907587-4c4cea80-325c-11e9-94a9-c02673c9cb23.png)

So "butt" contains the mustache code necessary to render 'a', but it doesn't...

**Comment by liambaloh** [comment]
Ah! Just as I thought! I had to make a public post asking for help in order to figure it out! 

Turns out everything is explained here: https://github.com/bobthecow/mustache.php/wiki#using-all-these-options

Partial definitions have to be defined elsewhere - during the instantiation of the mustache engine:
![image](https://user-images.githubusercontent.com/1364475/52907696-7a332e80-325e-11e9-9862-8417a0c436ae.png)

![image](https://user-images.githubusercontent.com/1364475/52907700-89b27780-325e-11e9-81a7-fc811ab088cd.png)

![image](https://user-images.githubusercontent.com/1364475/52907701-9040ef00-325e-11e9-843f-fa7f9da7cf06.png)

![image](https://user-images.githubusercontent.com/1364475/52907703-98992a00-325e-11e9-857e-878292dc80ed.png)

Success!

**Comment by mkalam-alami** [comment]
Nice job it seems like you got a lot of stuff done :)

If you're interested, to prevent having each partial listed explicitly you can use a [FileSystem loader](https://github.com/bobthecow/mustache.php/wiki/Template-Loading#filesystem-loader) to let Mustache know where your templates & partials are.
